src/reprepro_updater/repository_info.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+
+class RepositoryInfo:
+    """
+    Parse apt repository data in order to provide information for reprepro-updater.
+    """
+    def __init__(self, repo_dir, distro, arch):
+        self.repo_dir = repo_dir
+        self.distro = distro
+        self.arch = arch
+        self.package_dependencies = None
+        self.package_rdepends = None
+
+    def _parse_packages_file(self):
+        """
+
+        Loads the Packages file for the main repository component and parses
+        the dependencies into memory.
+
+        Sets the variable package_dependencies which is a dict of package names
+        mapping to sets of dependency names.
+        """
+        if self.package_dependencies is None:
+            self.package_dependencies = {}
+        packages_filepath = os.path.join(self.repo_dir, 'dists', self.distro,
+                                         'main', 'binary-{}'.format(self.arch), 'Packages')
+        packages_contents = open(packages_filepath, "r").read()
+        for section in packages_contents.split("\n\n"):
+            if section is '':
+                continue
+            name = None
+            depends = None
+            for line in section.split("\n"):
+                if line.startswith("Package: "):
+                    name = line.split(": ")[1]
+                if line.startswith("Depends: "):
+                    depends = line.split(": ")[1]
+            if name is None:
+                raise RuntimeError(
+                    "Repository file '{}' had a section missing 'Package':\n+\n{}\n+".format(
+                        packages_filepath, section))
+            if depends is None:
+                dependency_set = set()
+            else:
+                dependency_set = set([item.strip().split(' ')[0] for item in depends.split(',')])
```

Comment:
I would suggest to perform this set comprehension right after the value is extracted from the file (line 38). By initializing `depends` in the beginning you can then get rid of the second variable `dependency_set`.